### PR TITLE
Use fontc as alternative gftools.builder compiler

### PR DESCRIPTION
**Pull request by simoncozens**
Implements #1045, if you set `GFTOOLS_COMPILER=fontc` environment variable. Some caveats:

- Any `args` that invoke `--filter` are dropped on the floor. Ideally these should be converted, ie. `--filter FlattenComponentsFilter` should become `--flatten-components`, etc.
- `instantiateUFO` -> `buildTTF` pipelines won't quite work yet as we assume JSON UFO structures in instantiated UFOs, which `fontc` doesn't support. Maybe we could make the JSON-ness dependent on `FontmakeOperationBase.compiler != "fontc"`.

### CODE REVIEW

File: Lib/gftools/builder/operations/__init__.py [hunk 1]
```diff
@@ -131,9 +143,24 @@ def validate(self):
 
 
 class FontmakeOperationBase(OperationBase):
+    COMPILER_ENV_KEY = "GFTOOLS_COMPILER"
```
Comment (cmyr) [inline Lib/gftools/builder/operations/__init__.py:146]: couldn't we just have a different subclass of `OperationBase`, and switch to that dynamically? then we'd avoid junking up this class with fontc stuff, and I could properly handle the arguments...

Comment (simoncozens) [reply Lib/gftools/builder/operations/__init__.py:146]: If you believe it would be cleaner, I would be happy to see it.

Comment (cmyr) [reply Lib/gftools/builder/operations/__init__.py:146]: The only part I'm not sure about is how best to swap the classes in the presence of the CLI flag. Do you have any ideas?

Comment (simoncozens) [reply Lib/gftools/builder/operations/__init__.py:146]: Nope, not really. Once a class is loaded in Python it's unpleasant to try to change its base class. This is why I suggested doing things the way that I did ...

Comment (rsheeter) [reply Lib/gftools/builder/operations/__init__.py:146]: I'm not following why we'd need to change base, we should have two different classes and pick which one to create based on cli. I don't know the gftools codebase, is there something about it that makes that difficult?

Comment (simoncozens) [reply Lib/gftools/builder/operations/__init__.py:146]: If you're not changing base class, then you will need six different classes (one implementation each of `buildTTF`, `buildOTF`, `buildVariable` for fontc and fontbakery). This is *less* elegant than the solution I'm proposing.

Comment (rsheeter) [reply Lib/gftools/builder/operations/__init__.py:146]: I think I would dispute the elegance of writing `if self.compiler ==` :)

**Pull request closed without merging**